activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -107,7 +107,8 @@ def take!
     end
 
     # Find the first record (or first N records if a parameter is supplied).
-    # If no order is defined it will order by primary key.
+    # If no order is defined it will order by primary key if the collection is not loaded.
+    # If the collection is loaded, the order can be non-deterministic.
```

Comment:
That is not true. The collection can have a deterministic order depending on the relation. Maybe `If the collection is loaded it will get the first item of that collection.`